requirements-dev.txt

```
@@ -8,7 +8,7 @@
 bandit[toml] >=1.0, <2.0
 black >=23.0, <24.0
 cffconvert >=2.0, <3.0
-flake8 >=6.0, <7.0
+flake8 >=6.0, <8.0
```

```suggestion
flake8 >=7.0, <8.0
```

Also, update the flake8 pre-commit hook.

The options `--include-in-doctest` and `--exclude-from-doctest` that allowed the user to specify the files that should be checked by PyFlakes for doctest syntax (https://flake8.pycqa.org/en/6.1.0/user/options.html#cmdoption-flake8-include-in-doctest) have been removed in version 7.0.0 of flake8 (https://flake8.pycqa.org/en/latest/release-notes/7.0.0.html#backwards-incompatible-changes).  Make sure that we do not use them anymore (e.g. in the `.flake8` config file or in the `test.yml` CI file).